lib/elastic_apm/transport/worker.rb

```diff
@@ -17,15 +25,15 @@ def initialize(
         queue,
         serializers:,
         filters:,
-        conn_adapter: Connection
+        conn_adapter: nil
```

Comment:
I don't think we should have both methods of Dependency Injection. IMO let's remove `conn_adapter`. It's only used in `worker_spec.rb`.